src/components/MiniWidgetInstantiator.vue

```diff
@@ -3,28 +3,40 @@
 </template>
 
 <script setup lang="ts">
-import { type AsyncComponentLoader, defineAsyncComponent, toRefs } from 'vue'
+import { defineAsyncComponent, onMounted, toRefs } from 'vue'
 
-import type { MiniWidget, MiniWidgetType } from '@/types/widgets'
+import { createCockpitActionVariable } from '@/libs/actions/data-lake'
+import type { CustomWidgetElement, MiniWidget } from '@/types/widgets'
 
 const props = defineProps<{
   /**
-   * Mini-widget instance
+   *
    */
-  miniWidget: MiniWidget
+  miniWidget: MiniWidget | CustomWidgetElement
 }>()
 
 const miniWidget = toRefs(props).miniWidget
 
-const componentCache: Record<string, AsyncComponentLoader> = {}
+const componentCache: Record<string, ReturnType<typeof defineAsyncComponent>> = {}
 
-const componentFromType = (componentType: MiniWidgetType): AsyncComponentLoader => {
-  if (componentCache[componentType] === undefined) {
-    componentCache[componentType] = defineAsyncComponent(
-      () => import(`../components/mini-widgets/${componentType}.vue`)
-    )
+const componentFromType = (componentType: string): ReturnType<typeof defineAsyncComponent> => {
+  if (!componentCache[componentType]) {
+    componentCache[componentType] = defineAsyncComponent(async () => {
+      try {
+        return await import(`../components/mini-widgets/${componentType}.vue`)
+      } catch (error) {
+        return await import(`../components/custom-widget-elements/${componentType}.vue`)
+      }
+    })
   }
-
   return componentCache[componentType]
 }
+
+const registerCockpitActions = (): void => {
+  if (miniWidget.value.options.actionVariable) {
+    createCockpitActionVariable(miniWidget.value.options.actionVariable)
+  }
+}
+
+onMounted(() => registerCockpitActions())
```

Comment:
One question: this being in the MiniWidgetInstantiator, it means the variable will not be created unless the mini-widget is instantiated? Should we move this registration to the `widgetManager.ts` store, so that every widget that exists in the profile will have its variables created? I trully don't remember if all mini-widgets that are in the profile will be instantiated or not on boot, so I may be wrong here.

Reply:
As I could observe, every mini-widget will be instantiated on boot by the edit menu mini-widget cards and also if they exists on any view, they will be instantiated even if the view is not active.

Since the variables are particularly related to its mini-widget, I think its a good place to register the variable on the data-lake.
Do you prefer by a architectural matter, to make this registration on the widgetManager.ts?